controls/tests.py

```diff
@@ -895,3 +895,107 @@ def create_simple_import_record(self):
         statement.save()
 
         return import_record
+
+class ImportExportProjectTests(OrganizationSiteFunctionalTests):
+    """
+    Testing the whole import of project JSON which will form a new system, project, and set of components and their statements.
+    """
+
+    def test_new_project_json_import(self):
+        """
+        Testing the import of a new project through JSON
+        """
+
+        # login as the first user and create a new project
+        self._login()
+        self._new_project()
+
+        # Checks the number of projects and components before the import
+        self.assertEqual(Project.objects.all().count(), 3)
+        self.assertEqual(Element.objects.all().exclude(element_type='system').count(), 0)
+
+        ## Import a new project
+        # click import project button, opening the modal
+        self.click_element("#action-buttons\ action-row > div.btn-group > button:nth-child(4)")
+        ## select through the modal information needed and browse for the import needed
+        self.select_option_by_visible_text("#id_appsource_compapp", "project")
+        self.browser.find_element_by_xpath("(//option[@value='3'])[2]").click()
+        self.browser.find_element_by_id("id_importcheck").click()
+        #self.browser.find_element_by_id("id_file").click()
+        file_input = self.browser.find_element_by_css_selector("#id_file")
+
+        file_path = os.getcwd() + "/fixtures/test_project_import_data.json"
+        # convert filepath if necessary and send keys
+        self.filepath_conversion(file_input, file_path, "sendkeys")
+        self.browser.find_element_by_id("import_component_submit").click()
+
+        # Check the new number of projects, and validate that it's 1 more than the previous count.
+        self.assertEqual(Project.objects.all().count(), 4)
+        # Has the correct name?
+        self.assertEqual(Project.objects.get(id=4).title, "New Test Project")
+        # Components and their statements?
+        self.assertEqual(Element.objects.all().exclude(element_type='system').count(), 1)
+        self.assertEqual(Statement.objects.all().count(), 3)
+
+    def test_update_project_json_import(self):
+        """
+        Testing the update of a project through project JSON import and ingestion of components and their statements
+        """
+
+        # login as the first user and create a new project
+        self._login()
+        self._new_project()
+
+        # Checks the number of projects and components before the import
+        self.assertEqual(Project.objects.all().count(), 3)
```

Comment:
My concern here is if other tests add (or delete) more projects in the future, then this "3" won't be correct and the test will (wrongly) break. We might have to consider storing the "pre-import" value in a variable, and then doing an assert for the difference afterwards, like:
`project_count_before_import = Project.objects.all().count()`
`# Do import stuff`
`self.assertEqual(Project.objects.all().count(), project_count_before_import + 1)`

That would be the case with Elements and Statements. It's happening throughout the tests, so that's probably some technical debt we'll have to resolve down the road throughout the project.